internal/datacoord/garbage_collector.go

```diff
@@ -134,6 +139,16 @@ func (gc *garbageCollector) scan() {
 				v++
 				continue
 			}
+			var keyIsReferenced bool
+			for _, segID := range segReferIDs {
+				if keyIsReferenced = strings.Contains(info.Key, strconv.FormatInt(segID, 10)); keyIsReferenced {
```

Comment:
I don't understand.
1) for garbage collector, it shouldn't use listobjects, but should follow exactly what etcd meta has, we can leave the fix to next patch, otherwise this is very easy to inconsistent.
2) even if we list objects, we should always parse segment id from info.key but not use string.contains.There should be utility to do so